### PR TITLE
Add unavailable pathWithComponents method for warning

### PR DESCRIPTION
**Pull request by keith**

### CODE REVIEW

**Comment by keith** [comment]
I submitted this because of the comment here: https://github.com/apple/swift-corelibs-foundation/pull/2290#issuecomment-495783466

**Comment by keith** [comment]
@millenomi are you still interested in this?

File: Foundation/NSStringAPI.swift [hunk 1]
```diff
@@ -1894,6 +1894,14 @@ extension StringProtocol {
 
 }
 
+// Old method names
+extension NSString {
+    @available(*, unavailable, message: "Use path(withComponents:) instead.")
```
Comment (spevans) [inline Foundation/NSStringAPI.swift]: Should this use `renamed: path(withComponents:)` to get a fix-it instead of `message: ...`?

Comment (keith) [reply Foundation/NSStringAPI.swift]: Updated!

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by keith** [comment]
@spevans looks like it didn't trigger, mind trying again?

**Comment by spevans** [comment]
@swift-ci test

**Comment by keith** [comment]
@spevans looks like this one is green now!

**Comment by keith** [comment]
@Bigcheese friendly ping

**Comment by keith** [comment]
@spevans friendly ping

**Comment by spevans** [comment]
@keith It looks like this needs to be rebased to fix some conflicts

**Comment by keith** [comment]
Sorry, updated

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by keith** [comment]
@spevans is this good to go now?

**Comment by spevans** [comment]
@swift-ci test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci test linux

**Comment by keith** [comment]
Thanks!

**Pull request merged**